lib/tests/streamlit/elements/echo_test.py

```diff
@@ -70,10 +69,71 @@ def do_y(self):
 
         self.clear_queue()
 
+    @parameterized.expand(
+        [
+            ("code_location default", {}, 0, 1),
+            ("code_location above", {"code_location": "above"}, 0, 1),
+            ("code_location below", {"code_location": "below"}, 1, 0),
+        ]
+    )
+    def test_echo_unindent(
+        self,
+        _,
+        echo_kwargs_very_long_name_very_long_very_very_very_very_very_very_long,
+        echo_index,
+        output_index,
+    ):
+        with st.echo(
+            **echo_kwargs_very_long_name_very_long_very_very_very_very_very_very_long
+        ):
+            st.write("Hello")
+            "hi"
+
+            def foo(x):
+                y = x + 10
+
+                print(y)
+
+            class MyClass(object):
+                def do_x(self):
+                    pass
+
+                def do_y(self):
+                    pass
+
+        echo_str = """st.write("Hello")
+"hi"
+
+def foo(x):
+    y = x + 10
+
+    print(y)
+
+class MyClass(object):
+    def do_x(self):
+        pass
+
+    def do_y(self):
+        pass"""
+
+        element = self.get_delta_from_queue(echo_index).new_element
```

Comment:
Tests using `get_delta_from_queue` could also be written with the interactive script tests framework, which might be clearer, though this is not required.